File: src/main/webapp/storage/src/app/dto/Sell.ts
import {MovingRecord} from "./MovingRecord";

export class Sell {
  private readonly _id: number;
  private readonly _date: string;
  private readonly _username: string;
  private readonly _items: MovingRecord[];

  constructor(id: number, date: string, username: string, items: MovingRecord[]) {
    this._id = id;
    this._date = date;
    this._username = username;
    this._items = items;
  }

  get id(): number {
    return this._id;
  }

  get date(): string {
    return this._date;
  }

  get items(): MovingRecord[] {
    return this._items;
  }

  get username(): string {
    return this._username;
  }
}

File: src/main/java/com/sonet/storage/model/sell/Sell.java
package com.sonet.storage.model.sell;

import com.sonet.storage.model.moving.MovingRecord;
import com.sonet.storage.model.user.User;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "sell")
@NoArgsConstructor
@Getter
@Setter
public class Sell {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String date;

    @ManyToOne
    private User user;

    @OneToMany
    private List<MovingRecord> items;

    public Sell(String date, List<MovingRecord> items) {
        this.date = date;
        this.items = items;
    }
}
